lib/src/visitor/async_evaluate.dart

```diff
@@ -2319,11 +2469,78 @@ final class _EvaluateVisitor
       // The simplification logic in the [SassCalculation] static methods will
       // throw an error if the arguments aren't compatible, but we have access
       // to the original spans so we can throw a more informative error.
-      _verifyCompatibleNumbers(arguments, node.arguments);
+      if (error.message.contains("compatible")) {
+        _verifyCompatibleNumbers(arguments, node.arguments.positional);
+      }
       throwWithTrace(_exception(error.message, node.span), error, stackTrace);
     }
   }
 
+  /// Verifies that the calculation [node] has the correct number of arguments.
+  void _checkCalculationArguments(FunctionExpression node) {
+    void check([int? maxArgs]) {
+      if (node.arguments.positional.isEmpty) {
+        throw _exception("Missing argument.", node.span);
+      } else if (maxArgs != null &&
+          node.arguments.positional.length > maxArgs) {
+        throw _exception(
+            "Only $maxArgs ${pluralize('argument', maxArgs)} allowed, but "
+                    "${node.arguments.positional.length} " +
+                pluralize('was', node.arguments.positional.length,
+                    plural: 'were') +
+                " passed.",
+            node.span);
+      }
+    }
+
+    switch (node.name.toLowerCase()) {
+      case "calc":
+        check(1);
+      case "sqrt":
+        check(1);
+      case "sin":
+        check(1);
+      case "cos":
+        check(1);
+      case "tan":
+        check(1);
+      case "asin":
+        check(1);
+      case "acos":
+        check(1);
+      case "atan":
+        check(1);
+      case "abs":
+        check(1);
+      case "exp":
+        check(1);
+      case "sign":
+        check(1);
+      case "min":
+        check();
+      case "max":
+        check();
+      case "hypot":
+        check();
+      case "pow":
+        check(2);
+      case "atan2":
+        check(2);
+      case "log":
+        check(2);
+      case "mod":
+        check(2);
+      case "rem":
+        check(2);
+      case "round":
+        check(3);
+      case "clamp":
+        check(3);
```

Comment:
These take a lot of vertical space, can you merge the cases with similar expected outcomes? I think you can either do:
```dart
switch (node.name.toLowerCase()) {
  case "calc" || "sqrt":
    check(1);
  // ... same for check 2 and check 3
  case "min" || "max":
    check();
  case _: // ...
}
```
or
```dart
switch (node.name.toLowerCase()) {
  // ...
  case "calc":
  case "sqrt":
    check(1);
  // ...
  case "min":
  case "max":
    check();
  case _: // ...
}
```